File: pset6/readability.py
from cs50 import get_string

# promt user for text
text = get_string("Text: ")

words = 1
sentences = 0
letters = 0

# check each character
for char in text:
    if char.isalpha():
        letters += 1
    if char.isspace():
        words += 1
    if char in ['?','!','.']:
        sentences += 1

# calculate letters and sentences over 100 words
av_letters = (letters * 100.0) / words
av_s = (sentences * 100.0) / words

# apply the index
index = int((0.0588 * av_letters - 0.296 * av_s - 15.8) + 0.5)

# print grade level
if index <1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>
#include <string.h>



int main(void)
{
    // promt user for text
    string text = get_string("Text: ");
    
    int words = 0, sentences = 0, letters = 0;
    
    // iterate over text checking each character
    for (int i = 0, n = strlen(text); i < n; i++)
    
    if (isalpha(text[i]))
    {
        letters++;
    }
 
    // count words
    else if (isspace(text[i]))
    {
        words++;
    }
    // count sentences
    else if ((text[i]) == '.' || (text[i]) == '?')
    {
        sentences++;
    }
    
    words = words + 1;
    
    float L = ((float) letters / (float)words) * 100 ;
    float S = ((float) sentences / (float) words) * 100;
    // calculate
    float grade = 0.0588 *  L - 0.296 * S - 15.8;
    
    int index = round(grade);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %i\n", index);
    }

  
}
